Flaky parser/grok tests

I get the following failure intermittently on my local machine with Node.js 4.2.4:

```
  1) parsers/grok stops emitting values after the optimization.limit:

      AssertionError: expected 1 to equal 2
      + expected - actual

      -1
      +2
      
      at test/adapters/parsers/grok.spec.js:125:42
      at bound (domain.js:287:14)
      at runBound (domain.js:300:12)
      at tryCatcher (node_modules/bluebird/js/main/util.js:26:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/main/promise.js:507:31)
      at Promise._settlePromiseAt (node_modules/bluebird/js/main/promise.js:581:18)
      at Promise._settlePromises (node_modules/bluebird/js/main/promise.js:697:14)
      at Async._drainQueue (node_modules/bluebird/js/main/async.js:123:16)
      at Async._drainQueues (node_modules/bluebird/js/main/async.js:133:10)
      at Immediate.Async.drainQueues [as _onImmediate] (node_modules/bluebird/js/main/async.js:15:14)



[14:03:55] 'test' errored after 1.3 min
[14:03:55] Error in plugin 'gulp-mocha'
Message:
    1 test failed.
```

--- a/test/adapters/parsers/csv.spec.js
+++ b/test/adapters/parsers/csv.spec.js
@@ -95,26 +95,26 @@ describe('parsers/csv', function() {
 
         return serializer.done()
         .then(() => {
-            stream.end();
+            stream.end(function() {
+                var csv = parsers.getParser('csv', {
+                    optimization: {
+                        type: 'head',
+                        limit: 1
+                    }
+                });
 
-            var csv = parsers.getParser('csv', {
-                optimization: {
-                    type: 'head',
-                    limit: 1
-                }
-            });
-
-            var results = [];
-            return csv.parseStream(fs.createReadStream(tmpFilename), function(result) {
-                results.push(result);
-            })
-            .then(function() {
-                expect(results.length).to.be.equal(2); // 1 point + empty batch
-                expect(csv.totalRead).to.be.lessThan(40);
-                expect(csv.totalParsed).to.equal(2); // we always parse one ahead
-            })
-            .finally(function() {
-                fs.unlinkSync(tmpFilename);
+                var results = [];
+                return csv.parseStream(fs.createReadStream(tmpFilename), function(result) {
+                    results.push(result);
+                })
+                .then(function() {
+                    expect(results.length).to.be.equal(2); // 1 point + empty batch
+                    expect(csv.totalRead).to.be.lessThan(40);
+                    expect(csv.totalParsed).to.equal(2); // we always parse one ahead
+                })
+                .finally(function() {
+                    fs.unlinkSync(tmpFilename);
+                });
             });
         });
     });

--- a/test/adapters/parsers/grok.spec.js
+++ b/test/adapters/parsers/grok.spec.js
@@ -97,7 +97,7 @@ describe('parsers/grok', function() {
         });
     });
 
-    withGrokIt('stops emitting values after the optimization.limit', function() {
+    withGrokIt('stops emitting values after the optimization.limit', function(done) {
         // the read ahead buffer of the parser will always read more points
         // that we actually want to parse but lets make sure this does not // read the whole file by writing out enough data
         var tmpFilename = tmp.tmpNameSync();
@@ -107,27 +107,28 @@ describe('parsers/grok', function() {
                          parseInt(Math.random()*64*1024) +
                          ']: ' + Array(1024).join('X') + '\n');
         }
-        stream.end();
+        stream.end(function() { 
+            var parser = parsers.getParser('grok', {
+                pattern: '%{SYSLOGLINE}',
+                optimization: {
+                    type: 'head',
+                    limit: 1
+                }
+            });
 
-        var parser = parsers.getParser('grok', {
-            pattern: '%{SYSLOGLINE}',
-            optimization: {
-                type: 'head',
-                limit: 1
-            }
-        });
-
-        var results = [];
-        return parser.parseStream(fs.createReadStream(tmpFilename), function(result) {
-            results.push(result);
-        })
-        .then(function() {
-            expect(results.length).to.be.equal(2); // 1 point + empty batch
-            expect(parser.totalRead).to.be.lessThan(500);
-            expect(parser.totalParsed).to.equal(2); // always parse one ahead
-        })
-        .finally(function() {
-            fs.unlinkSync(tmpFilename);
+            var results = [];
+            parser.parseStream(fs.createReadStream(tmpFilename), function(result) {
+                results.push(result);
+            })
+            .then(function() {
+                expect(results.length).to.be.equal(2); // 1 point + empty batch
+                expect(parser.totalRead).to.be.lessThan(500);
+                expect(parser.totalParsed).to.equal(2); // always parse one ahead
+            })
+            .finally(function() {
+                fs.unlinkSync(tmpFilename);
+                done();
+            });
         });
     });
 

